libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -765,6 +765,12 @@ void NavEKF2::UpdateFilter(void)
         }
     }
 
+    // when disarmed force lane1 if it is healthy. This is a temporary
+    // workaround for poor performance of the 2nd IMU
+    if (!hal.util->get_soft_armed() && primary != 0 && core[0].healthy()) {
```

Comment:
Looks good. But seems like it can still switch to the other EKF Lane in case the lane1 is not healthy, rather, not using any lane at all. Is this what we want, or should we deny switching to any lane completely in case the 1st lane is not ready when the vehicle is in soft_armed mode?

Reply:
What about this?